misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java

```diff
@@ -3315,6 +3277,136 @@ public BreakStatementNode transform(BreakStatementNode breakStatementNode) {
+                .withOpenBracket(openBracket)
+                .withCloseBracket(closeBracket)
+                .withMemberTypeDesc(memberTypeDesc)
+                .apply();
+
+    }
+
+    @Override
+    public XMLElementNode transform(XMLElementNode xMLElementNode) {
+        XMLStartTagNode startTagNode = formatNode(xMLElementNode.startTag(), 0, 0);
+        NodeList<XMLItemNode> content = formatNodeList(xMLElementNode.content(), 0, 0, 0, 0);
+        XMLEndTagNode endTagNode = formatNode(xMLElementNode.endTag(), this.trailingWS, this.trailingNL);
+
+        return xMLElementNode.modify()
+                .withStartTag(startTagNode)
+                .withContent(content)
+                .withEndTag(endTagNode)
+                .apply();
+    }
+
+    @Override
+    public XMLStartTagNode transform(XMLStartTagNode xMLStartTagNode) {
+        Token ltToken = formatToken(xMLStartTagNode.ltToken(), 0, 0);
+        int nameTrailingWS = xMLStartTagNode.attributes().isEmpty() ? 0 : 1;
+        XMLNameNode name = formatNode(xMLStartTagNode.name(), nameTrailingWS, 0);
+        NodeList<XMLAttributeNode> attributes = formatNodeList(xMLStartTagNode.attributes(), 1, 0, 0, 0);
+        Token getToken = formatToken(xMLStartTagNode.getToken(), this.trailingWS, this.trailingNL);
+
+        return xMLStartTagNode.modify()
+                .withLtToken(ltToken)
+                .withName(name)
+                .withAttributes(attributes)
+                .withGetToken(getToken)
+                .apply();
+    }
+
+    @Override
+    public XMLEndTagNode transform(XMLEndTagNode xMLEndTagNode) {
+        Token ltToken = formatToken(xMLEndTagNode.ltToken(), 0, 0);
+        Token slashToken = formatToken(xMLEndTagNode.slashToken(), 0, 0);
+        XMLNameNode name = formatNode(xMLEndTagNode.name(), 0,0);
+        Token getToken = formatToken(xMLEndTagNode.getToken(), this.trailingWS,this.trailingNL);
+
+        return xMLEndTagNode.modify()
+                .withLtToken(ltToken)
+                .withSlashToken(slashToken)
+                .withName(name)
+                .withGetToken(getToken)
+                .apply();
+    }
+
+    @Override
+    public XmlTypeDescriptorNode transform(XmlTypeDescriptorNode xmlTypeDescriptorNode) {
+        int trailingWS = xmlTypeDescriptorNode.xmlTypeParamsNode().isEmpty() ? this.trailingWS : 0;
+        int trailingNL = xmlTypeDescriptorNode.xmlTypeParamsNode().isEmpty() ? this.trailingNL : 0;
+        Token xmlKeywordToken = formatToken(xmlTypeDescriptorNode.xmlKeywordToken(), trailingWS, trailingNL);
+        if (xmlTypeDescriptorNode.xmlTypeParamsNode().isPresent()) {
+            TypeParameterNode xmlTypeParamsNode = formatNode(xmlTypeDescriptorNode.xmlTypeParamsNode().get(),
+                    this.trailingWS, this.trailingNL);
+            xmlTypeDescriptorNode = xmlTypeDescriptorNode.modify()
+                    .withXmlTypeParamsNode(xmlTypeParamsNode)
+                    .apply();
+        }
```

Comment:
Can we simplify this logic to something like [1]?

[1] https://github.com/ballerina-platform/ballerina-lang/blob/bdaa37b9f498322489b9241c3cdbd1d03b3376b6/misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java#L348

Reply:
Done.